controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java

```diff
@@ -3809,4 +3809,7 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
     @LogMessage(level = WARN)
     @Message(id = 517, value = "There are multiple Parallel Boot Operations.")
     void multipleParallelBootOperation();
+
+    @Message(id = 518, value = "Resource /extension=%s uses a non-canonical module name; use the canonical form %s")
```

Comment:
Perhaps, I would add in the message a short hint about what a canonical form is. Something like:

"Resource /extension=%s uses a non-canonical module name; use the canonical form %s. The canonical representation includes slot information in the module name if the slot is not 'main'"

Reply:
Done.